lib/resolver.js

```diff
@@ -78,7 +78,10 @@ Resolver.prototype = {
     var themeComponent = this.themeComponentFor(scssFile, type, { src: this.stylesDir, target: this.componentsDir });
     var themeDefaultComponent = this.themeDefaultComponentFor(themeComponent, type, { src: this.componentsDir, target: this.componentsDir });
 
-    var parentThemeScssFile = scssFile.replace(context.prefix + '--', '').replace(themeName, context.parentThemeName);
+    var parentThemeScssFile = scssFile
+      .replace(context.prefix + '--', parentThemeContext.prefix + '--')
```

Comment:
Instead of removing the subtheme's prefix, we replace it with the parent theme's prefix